torcharrow/test/integration/test_criteo.py

```diff
@@ -0,0 +1,217 @@
+import torcharrow.pytorch as tap
+from torcharrow.functional import functional
+
+try:
+    import pyarrow.parquet as pq
+
+    HAS_PYARROW_PARQUET = True
+except ImportError:
+    HAS_PYARROW_PARQUET = False
+skipIfNoPyArrowParquet = unittest.skipIf(not HAS_PYARROW_PARQUET, "no PyArrow Parquet")
+
+# Based on https://github.com/facebookresearch/torchrec/blob/9a8a4ad631bbccd7cd8166b7e6d7607e2560d2bd/torchrec/datasets/criteo.py#L37-L46
+INT_FEATURE_COUNT = 13
+CAT_FEATURE_COUNT = 26
+DEFAULT_LABEL_NAME = "label"
+DEFAULT_INT_NAMES: List[str] = [f"int_{idx}" for idx in range(INT_FEATURE_COUNT)]
+DEFAULT_CAT_NAMES: List[str] = [f"cat_{idx}" for idx in range(CAT_FEATURE_COUNT)]
+
+DTYPE = dt.Struct(
+    [
+        dt.Field(DEFAULT_LABEL_NAME, dt.int8),
+        dt.Field(
+            "dense_features",
+            dt.Struct(
+                [
+                    dt.Field(int_name, dt.Int32(nullable=True))
+                    for int_name in DEFAULT_INT_NAMES
+                ]
+            ),
+        ),
+        dt.Field(
+            "sparse_features",
+            dt.Struct(
+                [
+                    dt.Field(cat_name, dt.Int32(nullable=True))
+                    for cat_name in DEFAULT_CAT_NAMES
+                ]
+            ),
+        ),
+    ]
+)
+
+# TODO: allow to use to_tensor with Callable
+# TODO: implement conversion in native C++
+class _DenseConversion(tap.ITorchConversion):
+    def to_tensor(self, df: ta.IDataFrame):
+        # Default to_tensor, each field is a Tensor
+        tensors = df.to_tensor()
+
+        # Stack them into BatchSize * NumFields Tensor
+        return torch.cat(
+            [column_tensor.values.unsqueeze(0).T for column_tensor in tensors], dim=1
+        )
+
+
+# Based on https://github.com/facebookresearch/torchrec/blob/main/torchrec/datasets/criteo.py#L441-L456
+# TODO: this is not a general purpose JaggedTensor conversion -- it leverages the fact that in Criteo preproc, each array is single element
+# TODO: implement general purpose jagged sparse tensor conversion in native C++
+class _CriteoJaggedTensorConversion(tap.ITorchConversion):
+    def to_tensor(self, df: ta.IDataFrame):
+        # TODO: Implement df.size(), similar to Pandas
+        num_arrays = len(df) * len(df.columns)
+
+        keys: List[str] = df.columns
+        lengths: torch.Tensor = torch.ones(
+            (num_arrays,),
+            dtype=torch.int32,
+        )
+        length_per_key: List[int] = len(df.columns) * [len(df)]
+
+        # Generate values
+        tensors = df.to_tensor()
+        # Stack them into BatchSize * NumFields Tensor
+        packed_tensor = torch.cat(
+            [
+                column_tensor.values.values.values.unsqueeze(0).T
+                for column_tensor in tensors
+            ],
+            dim=1,
+        )
+        # Flatten the tensors into one-dimension with columnar ordering
+        values = packed_tensor.transpose(1, 0).reshape(-1)
```

Comment:
Since you are going to flatten them, why not `values = torch.cat([column_tensor.values.values.values.values for column_tensor in tensors])`?

And I'm really confused by the concept of `column_tensor.values.values.values`

Reply:
> And I'm really confused by the concept of column_tensor.values.values.values

Yeah , this is quite confusing at this moment -- makes the default `to_tensor` difficult to use :(

The original column type is `List[Integer]`. Currently the type is incorrectly marked as "nullable" for both list and the internal integer. 

For the nullable type, it will the `to_tensor` result will be wrapped in `WithPresence`: https://github.com/facebookresearch/torcharrow/blob/4b5e8258e0e1f573a270dced7f26f3b5cf8de587/torcharrow/_pytorch/common.py#L17-L21

So there are to `.values` call required to get unwrap for `WithPresence`.

The last `.values` is to get the "flatten" array from `PackedList`, which is required: 
https://github.com/facebookresearch/torcharrow/blob/4b5e8258e0e1f573a270dced7f26f3b5cf8de587/torcharrow/_pytorch/common.py#L25-L29

Let me see if I can simplify it a bit (e.g. hack to make the column non-nullable~)

Reply:
> Since you are going to flatten them, why not values = torch.cat([column_tensor.values.values.values.values for column_tensor in tensors])?

I am copying the code from TorchRec repo ... -- `cat` by default will flatten all tensors, right?

Reply:
It will concatenate the first dimension by default. If inputs are in shape of `(10, 1)`, the output is going to be `(20, 1)`.
And, `(10)` will be `(20)`

Reply:
> The original column type is list[int32]. Currently the type is incorrectly marked as "nullable" for both list and the internal int32.

> For the nullable type, it will the to_tensor result will be wrapped in WithPresence:

I think it's fine to be nullable. But, whenever `to_tensor` is called, it should return a Tensor. And, depending on the tensor type, if it's dense tensor, we should raise Error for null value. If it's specified as sparse tensor, we may allow null value in Tensor

Reply:
The idea is if it's a nullable int, it will return with presence(mask) Tensor by default.  The problem is `functional` now pessimistically tag everything as nullable. -- will try to fix this separately 

We do plan to have dedicated `Dense` and `Jagged Sparse` conversion . So we will remove the two collation methods anyway.